core-profile-tck/jsonp-tck-ext/src/main/java/ee/jakarta/tck/core/jsonp/arquillian/MavenTestDependenciesDeploymentPackager.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) "2022" Red Hat and others
+ *
+ * This program and the accompanying materials are made available under the
+ * Apache Software License 2.0 which is available at:
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package ee.jakarta.tck.core.jsonp.arquillian;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Stream;
+
+import org.jboss.arquillian.container.spi.client.deployment.DeploymentDescription;
+import org.jboss.arquillian.container.test.spi.client.deployment.DeploymentScenarioGenerator;
+import org.jboss.arquillian.test.spi.TestClass;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.jboss.shrinkwrap.resolver.api.maven.Maven;
+import org.jboss.shrinkwrap.resolver.api.maven.MavenResolvedArtifact;
+import org.jboss.shrinkwrap.resolver.api.maven.ScopeType;
+
+/**
+ * Ideas copied from the Jakarta Batch MavenTestDependenciesDeploymentPackager to package up
+ * the test and compile scope artifacts in the current pom.xml. Exclude artifacts like the
+ * Jakarta EE artifacts by declaring them with provided scope.
+ */
+public class MavenTestDependenciesDeploymentPackager implements DeploymentScenarioGenerator {
+
+    @Override
+    public List<DeploymentDescription> generate(TestClass testClass) {
+        final WebArchive archive = generateDeployment();
+        final DeploymentDescription deployment = new DeploymentDescription("jsonp-tck-tests",
+                                                                                  archive);
+        return Collections.singletonList(deployment);
+    }
+
+    /**
+     * This resolves all test and compile scope artifacts in the project pom.xml and includes them as
+     * libraries in a jsonp-tck-tests-all.war
+     *
+     * @return shrinkwrap web archive for the test artifacts
+     */
+    private WebArchive generateDeployment() {
+        String[] activeMavenProfiles = {"staging"};
+        MavenResolvedArtifact[] resolvedArtifacts = Maven.resolver().loadPomFromFile("pom.xml", activeMavenProfiles)
+                .importDependencies(ScopeType.COMPILE, ScopeType.TEST)
+                .resolve()
+                .withTransitivity()
+                .asResolvedArtifact();
+
+        WebArchive archive = ShrinkWrap.create(WebArchive.class, "jsonp-tck-tests-all.war").as(WebArchive.class);
```

Comment:
> But the core profile does not support servlet

I've always felt we're making our lives more difficult by officially not supporting Servlet, but then practically assuming Servlet anyway.

Maybe we should clearly state why we think we dislike Servlet, and why we need to officially avoid it. Based on that we may want to discuss if a servlet-lite is feasible, which excludes all the things we think we need to avoid.

Reply:
We talked a bit about this at the end of the last platform call. We don't have a pure LibraryContainer archive type in Arquillian that can be used, so WebArchive is used as a proxy. We had to do this in CDI Lite as well. This is something that needs to be address going forward.

Restful does not support servlet either, but its TCK is what is bringing in the requirement to use a servlet. One aspect is wanting a lighter servlet api, but the other is allowing for the possibility to use another invocation mechanism. As of now, supporting servlets can be considered a TCK porting package implementation detail. We will need to address this in the future if a compatible implementation challenges this aspect of the TCK.

Reply:
> Restful does not support servlet either

If you mean Jakarta RESTful Web Services, then the Servlet 2.5 container was the only mandatory environment that needs to be supported by each implementation. That is why the TCK chose to use it. See https://jakarta.ee/specifications/restful-ws/3.1/jakarta-restful-ws-spec-3.1.html#servlet. Since 3.1 there is another possibility for SE Bootstrap, too.

Reply:
^ sounds like an itch for some implementations (as some do support war file deployment and some don't).  If I understand the above correctly, no one is against addressing this issue in the future but for EE 10, if there are practical suggestions, please do make them.  

Imo, it is time to merge this (later today) and setup CI jobs to build/stage the Core Profile TCK so implementations can start running against it.

Reply:
I think this at least allows the runtimes that support servlet execute the json-p and json-b tests.

Reply:
I think @starksm64 demonstrates how to get them running in a container, which is useful for EE container as a means to execute the tests.

Reply:
IMO, we should have consistent approach for JSONP TCK running in a container or as a standalone for all of Jakarta EE profiles.

Reply:
For EE10 the standalone approach is all that is required. The in container module is just an example for future work or for containers to experiment with.